distributed/comm/ucx.py

```diff
@@ -39,6 +43,50 @@ def init_once():
     import ucp as _ucp
 
     ucp = _ucp
+
+    # configuration of UCX can happen in two ways:
+    # 1) high level on/off flags which correspond to UCX configuration
+    # 2) explicity defined UCX configuration flags
+
+    # if any of the high level flags are set, as long as they are not Null/None,
+    # we assume we should configure basic TLS settings for UCX
+
+    if any(
+        [
+            dask.config.get("ucx.nvlink"),
+            dask.config.get("ucx.infiniband"),
+            dask.config.get("ucx.tcp-over-ucx"),
+            dask.config.get("ucx.net-devices"),
+        ]
+    ):
+        tls = "tcp,sockcm,cuda_copy"
+        tls_priority = "sockcm"
+
+        if dask.config.get("ucx.infiniband"):
+            tls = "rc," + tls
+        if dask.config.get("ucx.nvlink"):
+            tls = tls + ",cuda_ipc"
+
+        options = {"TLS": tls, "SOCKADDR_TLS_PRIORITY": tls_priority}
+
+        net_devices = dask.config.get("ucx.net-devices")
+        if net_devices is not None and net_devices != "":
+            options["NET_DEVICES"] = net_devices
+
+        dask.config.set({"ucx": options})
```

Comment:
Should we be concerned about losing the user's provided options here?

```python
In [1]: import dask                                                             

In [2]: dask.config.set({"a": {1: 1}})                                          
Out[2]: <dask.config.set at 0x7f25d2507610>

In [3]: dask.config.set({"a": {2: 2}})                                          
Out[3]: <dask.config.set at 0x7f25d256d610>

In [4]: dask.config.get("a")                                                    
Out[4]: {2: 2}
```

Maybe we want `dask.config.update` here instead?

Reply:
Well, what we want to avoid is passing in options which are valid for ucp.   ucp.init({'foo': 'bar'}) throws a `UCXConfigError`.  So either we want to remove invalid options or passing in a "clean' dictionary to `ucp.init`